test/codeu/chat/TestRunner.java

```diff
@@ -17,23 +17,85 @@
 import org.junit.runner.JUnitCore;
 import org.junit.runner.Result;
 import org.junit.runner.notification.Failure;
+import java.io.*;
 
 public final class TestRunner {
-  public static void main(String[] args) {
-     final Result result =
-         JUnitCore.runClasses(
-             codeu.chat.common.SecretTest.class,
-             codeu.chat.common.UuidTest.class,
-             codeu.chat.common.UuidsTest.class,
-             codeu.chat.relay.ServerTest.class,
-             codeu.chat.server.BasicControllerTest.class,
-             codeu.chat.server.DatabaseTest.class,
-             codeu.chat.server.RawControllerTest.class,
-             codeu.chat.util.store.StoreTest.class
-         );
-      for (final Failure failure : result.getFailures()) {
-         System.out.println(failure.toString());
+
+  private static boolean loading = true;
+  private static int frameSpeed = 60;
+
+  /**
+   * updated to a different thread as not to interfere
+   * with the loading message animation
+   *
+   * @param msg: the message to be displayed in the loading
+   */
+  public static synchronized void main(String[] args) {
+    Thread th = new Thread() {
+      @Override
+      public void run() {
+        try {
+          loading("Testing In Progress ... ");
+          final Result result =
+              JUnitCore.runClasses(
+                  codeu.chat.common.SecretTest.class,
+                  codeu.chat.common.UuidTest.class,
+                  codeu.chat.common.UuidsTest.class,
+                  codeu.chat.relay.ServerTest.class,
+                  codeu.chat.server.BasicControllerTest.class,
+                  codeu.chat.server.DatabaseTest.class,
+                  codeu.chat.server.RawControllerTest.class,
+                  codeu.chat.util.store.StoreTest.class
+              );
+          loading = false;
+          Thread.sleep(frameSpeed*2);
+          int numFailures = 0;
+          for (final Failure failure : result.getFailures()) {
+            numFailures++;
+          }
+          String results = "\rResult:                     "
+              + (result.wasSuccessful() ? "\n\tPassed\n"
+              : "\n\tFailed " + numFailures + " test(s):\n");
```

Comment:
`String.format()` might be useful here, along with a simple conditional statement for
`if (result.wasSeccussful()) {...`